cmd/nerdctl/compose.go

```diff
@@ -81,136 +64,33 @@ func newComposeCommand() *cobra.Command {
 	return composeCommand
 }
 
-func getComposer(cmd *cobra.Command, client *containerd.Client, globalOptions *types.GlobalCommandOptions) (*composer.Composer, error) {
-	nerdctlCmd, nerdctlArgs := globalFlags(cmd)
-	projectDirectory, err := cmd.Flags().GetString("project-directory")
+func getComposeOptions(cmd *cobra.Command) (*composer.Options, error) {
+	options := &composer.Options{}
+	options.NerdctlCmd, options.NerdctlArgs = globalFlags(cmd)
+	var err error
+	options.ProjectDirectory, err = cmd.Flags().GetString("project-directory")
 	if err != nil {
 		return nil, err
 	}
-	envFile, err := cmd.Flags().GetString("env-file")
+	options.EnvFile, err = cmd.Flags().GetString("env-file")
 	if err != nil {
 		return nil, err
 	}
-	projectName, err := cmd.Flags().GetString("project-name")
+	options.Project, err = cmd.Flags().GetString("project-name")
 	if err != nil {
 		return nil, err
 	}
-	debugFull := globalOptions.DebugFull
-	snapshotter := globalOptions.Snapshotter
-	files, err := cmd.Flags().GetStringArray("file")
+	options.DebugPrintFull, err = cmd.Flags().GetBool("debug-full")
 	if err != nil {
 		return nil, err
 	}
-	insecure := globalOptions.InsecureRegistry
-	cniPath := globalOptions.CNIPath
-	cniNetconfpath := globalOptions.CNINetConfPath
-	hostsDirs := globalOptions.HostsDir
-	experimental := globalOptions.Experimental
-
-	o := composer.Options{
-		Project:          projectName,
-		ProjectDirectory: projectDirectory,
-		ConfigPaths:      files,
-		EnvFile:          envFile,
-		NerdctlCmd:       nerdctlCmd,
-		NerdctlArgs:      nerdctlArgs,
-		DebugPrintFull:   debugFull,
-		Experimental:     experimental,
-	}
```

Comment:
can we keep this first half not changed (e.g., `return o, nil` directly here)? This part is only syntax changes and we can just do the `split it into 2 functions`. It makes the PR easy to review.

Reply:
And the return type can be `composer.Options` (i.e., not a pointer)